nixos/modules/virtualisation/containerd.nix

```diff
@@ -13,10 +15,21 @@ in
 
     configFile = lib.mkOption {
       default = null;
-      description = "path to containerd config file";
+      description = ''
+       Path to containerd config file.
+       Setting this option will override any configuration applied by the settings option.
```

Comment:
Since this leaves room for user error, having `settings` set, I opine we don't need `configFile` other than for back-compat. Hence, maybe a deprecation + warning is in order?

I don't think there is a good way to continue to support both config formats (into eternity).

Reply:
I could remove the mkDefault for `configFile` and than users have to use `mkForce` to set `configFile`. This should be quite explicit.

Reply:
Sounds excellent! :+1: